### PR TITLE
fix:Resolve deprecation warning for klogr.New in Metrics Server

### LINKED ISSUE #5732: Metrics Server: `klogr.New` is deprecated use `textlogger.NewLogger` instead
### Proposal

https://github.com/kedacore/keda/blob/9a00d1e7c3fd1d5acc4c736f2ff9f943d6aa1c2b/cmd/adapter/main.go#L58

```
cmd/adapter/main.go:58:14: SA1019: klogr.New is deprecated: this uses a custom, out-dated output format. Use textlogger.NewLogger instead. (staticcheck)
var logger = klogr.New().WithName("keda_metrics_adapter")
```

Let's migrate to the new interface, there might be some changes in the config/parameters etc.

It might also help with issues like: https://github.com/kedacore/keda/issues/5139

### Use-Case

_No response_

### Is this a feature you are interested in implementing yourself?

Maybe

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by guopeng0**
Metrics Server: klogr.New is deprecated use textlogger.NewLogger instead

### Checklist

- [x] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #5732 

Relates to #

### CODE REVIEW

File: Dockerfile.adapter [hunk 1]
```diff
@@ -39,4 +39,4 @@ COPY --from=builder /workspace/bin/keda-adapter .
 USER 65532:65532
 
 
-ENTRYPOINT ["/keda-adapter", "--secure-port=6443", "--logtostderr=true", "--v=0"]
+ENTRYPOINT ["/keda-adapter", "--secure-port=6443", "--logtostderr=true", "--v=0", "--zap-log-level=info", "--zap-encoder=console"]
```
Comment (zroubalik) [inline Dockerfile.adapter:42]: Why do we need this change please?

File: cmd/adapter/main.go [hunk 2]
```diff
@@ -56,9 +56,7 @@ type Adapter struct {
 }
 
 // https://github.com/kedacore/keda/issues/5732
```
Comment (zroubalik) [inline cmd/adapter/main.go:58]: this line should be remove too

File: Dockerfile.adapter (hunk 1 above)
Comment (JorTurFer) [reply Dockerfile.adapter:42]: These parameters are set as deployment args indeed to be changed by users

File: cmd/adapter/main.go (hunk 2 above)
Comment (guopeng0) [reply cmd/adapter/main.go]: It’s been updated.

File: Dockerfile.adapter (hunk 1 above)
Comment (guopeng0) [reply Dockerfile.adapter]: It’s been updated.

File: config/metrics-server/deployment.yaml [hunk 3]
```diff
@@ -60,6 +60,9 @@ spec:
           - --logtostderr=true
           - --stderrthreshold=ERROR
           - --v=0
+          - --zap-log-level=info
```
Comment (zroubalik) [inline config/metrics-server/deployment.yaml:63]: why is this needed?

**Review by zroubalik — COMMENTED** [review]
Could you please update docs for logs? https://github.com/kedacore/keda/blob/main/BUILD.md#setting-log-levels

File: cmd/adapter/main.go [hunk 4]
```diff
@@ -56,9 +56,7 @@ type Adapter struct {
...
-//
-//nolint:staticcheck // SA1019: klogr.New is deprecated.
-var logger = klogr.New().WithName("keda_metrics_adapter")
+var setupLog = ctrl.Log.WithName("keda_metrics_adapter")
```
Comment (samsond) [inline cmd/adapter/main.go:58]: I understand that **klogr.New** is deprecated, but I was wondering if there's a specific reason for introducing setupLog instead of reusing **logger** variable?

Would it be possible to update the code to consistently use logger without introducing setupLog?

File: config/metrics-server/deployment.yaml (hunk 3 above)
Comment (guopeng0) [reply config/metrics-server/deployment.yaml]: done

File: cmd/adapter/main.go (hunk 4 above)
Comment (guopeng0) [reply cmd/adapter/main.go:58]: With this commit, we've addressed the deprecation warning and have also aligned the logging approach with that of the KEDA operator. We can further refine the logging strategy in the next steps, and if necessary, I will submit a new pull request for additional changes.

**Comment by aliaqel-stripe** [comment]
Hi! I'm going to try and shepherd this PR to completion but in a different fork
